site/en/docs/extensions/mv3/getstarted/tut-focus-mode/index.md

```diff
@@ -0,0 +1,374 @@
+the current tab.
+
+This permission is enabled when the user **_invokes_** the extension. In this case, the user invokes the
+extension by clicking on the extension action.
+
+{% Details %}
+{% DetailsSummary %}
+💡 **What other user interactions enable the activeTab permission?**
+{% endDetailsSummary %}
+
+Other user gestures that enable the activeTab permission are the following:
+
+- Pressing a keyboard shortcut combination.
+- Selecting a context menu item.
+- Accepting a suggestion from the omnibox.
+- Opening an extension popup.
+
+{% endDetails %}
+
+The `activetab` permission allows users to _purposefully_ choose to run the extension on the
+currently focused tab; this way, it protects the user’s privacy. Another benefit is that it does not
+trigger a [permission warning][doc-perms-warning].
+
+To use the activetab permission, add it to the manifest's permission array:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "permissions": ["activeTab"],
+  ...
+}
+```
+
+### Step 4: Track the state of the current tab {: #step-4 }
+
+After the user clicks on the extension action, the extension will check if the URL matches a
+documentation page. Next, it will check the state of the current tab and set the next state. Add the
+following code to `background.js`:
+
+{% Label %}background.js:{% endLabel %}
+
+```js
+  chrome.action.onClicked.addListener(async (tab) => {
+    if (tab.url.startsWith(extensions) || tab.url.startsWith(webstore)) {
+      // Retrieve the action badge to check if the extension is 'ON' or 'OFF'
+      const prevState = await chrome.action.getBadgeText({ tabId: tab.id });
+      // Next state will always be the opposite
+      const nextState = prevState === 'ON' ? 'OFF' : 'ON'
+  
+      // Set the action badge to the next state
+      await chrome.action.setBadgeText({
+        tabId: tab.id,
+        text: nextState,
+      });
+  ...
+```
+
+### Step 5: Add or remove the stylesheet {: #step-5 }
+
+Now it's time to change the layout of the page. Create a file named `focus-mode.css` and include the
+following code:
+
+{% Label %}focus-mode.css:{% endLabel %}
+
+```js
+body > .scaffold > :is(top-nav, navigation-rail, side-nav, footer),
+main > :not(:last-child),
+main > :last-child > navigation-tree,
+main .toc-container {
+  display: none;
+}
+
+main > :last-child {
+  margin-top: min(10vmax, 10rem);
+  margin-bottom: min(10vmax, 10rem);
+}
+```
+
+Great! Let's insert or remove the stylesheet using the [Scripting][api-scripting] API. Start by
+declaring the `"scripting"` permission in the manifest:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "permissions": ["activeTab", "scripting"],
+  ...
+}
+```
+
+{% Aside 'success' %}
+
+The Scripting API does not trigger a [permission warning][doc-perms-warning].
+
+{% endAside %}
+
+Finally, in `background.js` add the following code to change the layout of the page:
+
+{% Label %}background.js:{% endLabel %}
+
+```js
```

Comment:
This entire code block is intended by two spaces. Is that intentional? If not, please remove the extra indentation.

Reply:
Unintentional, removing now.